tests/bsim/bluetooth/host/gatt/ccc_store/prj_2.conf

```diff
@@ -21,10 +21,5 @@ CONFIG_SETTINGS_NVS=y
 
 CONFIG_ASSERT=y
 
-# By doing so, BT_SETTINGS_DELAYED_STORE will be enabled because
-# BT_SETTINGS_CF_STORE_ON_WRITE is enabled by default and selecting it.
-# This cause the CCC to not be stored at all.
-# If BT_SETTINGS_CCC_LAZY_LOADING is enabled, the test will pass because the
-# CCC will not be cleared after the first disconnection and will then still be
-# in RAM.
+# Test that CCC is stored even if BT_SETTINGS_DELAYED_STORE is enabled
```

Comment:
didn't you mean 'disabled'?

Reply:
No, `BT_SETTINGS_DELAYED_STORE` is selected by `BT_SETTINGS_CCC_STORE_ON_WRITE` and `BT_SETTINGS_CCC_STORE_ON_WRITE`.
The problem here was that when `BT_SETTINGS_DELAYED_STORE` was enabled, in `bt_gatt_disconnected`, the behavior was to run the delayed store. But the delayed store operation was excluding the CCC or the CF if the store on write was enabled, leading to the issue.